### DISCUSSION TITLE
How did you manage visibility between routines?

### DISCUSSION THREAD

**Discussion by rFronteddu** [freeconf/restconf#42] [Q&A]
I was looking at your car example and I noticed that you do:

```
	for {
		<-time.After(time.Second)
		c.Miles += float64(c.Speed)
	}
```

And that Miles is consistently updated when I do a restconf get. Since I don't see any locking and I think the operations over Miles are done in separate routines, I was wondering if this is a side effect of my architecture and if the example shouldn't instead be done using locks to guarantee visibility between updates.

**Comment by dhubler** [comment]
Can you describe your issue in terms of what you expected, what you see
instead with a working example that demonstrates the issue?

re:locking
scalar values like float64 are non-volatile.  Absolute worst case you get a
value that is micro-seconds older.  If you really suspect

**Reply by rFronteddu** [reply]
Forgive me for the late reply.

I am not sure I follow. It is my understanding that unless you manually lock or use a channel, there is no guarantee that an update from a go routine will be visible from another. In this case, it may work due to hardware architecture but I don't think we should rely on that. Am I missing something?
